Zend/zend_string.c

```diff
@@ -392,32 +392,32 @@ ZEND_API bool ZEND_FASTCALL I_REPLACE_SONAME_FNNAME_ZU(NONE,zend_string_equal_va
 ZEND_API zend_never_inline NOIPA bool ZEND_FASTCALL zend_string_equal_val(const zend_string *s1, const zend_string *s2)
 {
 	const char *ptr = ZSTR_VAL(s1);
-	size_t delta = (const char*)s2 - (const char*)s1;
+	uintptr_t delta = (uintptr_t) s2 - (uintptr_t) s1;
```

Comment:
Just a question: what was wrong with ``size_t``.
Probably it would be better to use ``ssize_t`` (or ``offset_t``).
Now, may be ``intptr_t`` is better (I'm not sure).

Reply:
I think size_t is good too. On the x86 platforms they are the same size anyway. I just changed it to match the types of the right hand side without thinking too much about it. I can change it back to size_t.

Reply:
No problem. ``uintptr_t`` is good as well. (``size_t`` and pointers have the same size, according to some standard).